### ISSUE TITLE
In consistent of the result of "sample_n" between spark version and the local version 

### ISSUE THREAD

**Issue by TinyOcean** [sparklyr/sparklyr#669]
Hi,

I am dealing with some unbalance dataset, so I want to do under-sampling before getting to any models. 
Here are my code:

```
balance <- data_tbl %>% group_by(Y) %>% summarise(n=n()) %>% collect   
  if(max(balance$n) > unbalance.cutoff*min(balance$n)){   
    data_tbl %<>%   
      group_by(Y)  %>%   
      sample_n(size = min(balance$n)) %>%    
      ungroup  
  }
```

The original data looks like this:

> A tibble: 2 × 2
>       Y     n
>   <chr> <dbl>
> 1     0   554
> 2     1   246
> 

When running the above code in local mode, I got:

> A tibble: 2 × 2
>   Target     n
>    <int> <int>
> 1      0   246
> 2      1   246

While on sparklyr, the result is:

> A tibble: 2 × 2
>       Y     n
>   <chr> <dbl>
> 1     0   184
> 2     1    82

It is stratified sampling but not balance sampling.
So, how to perform undersampling in sparklyr? Thanks.

**Comment by TinyOcean** [comment]
My session Info:

> sessionInfo()
> R version 3.3.2 (2016-10-31)
> Platform: x86_64-apple-darwin13.4.0 (64-bit)
> Running under: OS X El Capitan 10.11.6
> 
> locale:
> [1] zh_CN.UTF-8/zh_CN.UTF-8/zh_CN.UTF-8/C/zh_CN.UTF-8/zh_CN.UTF-8
> 
> attached base packages:
> [1] stats     graphics  grDevices utils     datasets  methods   base     
> 
> other attached packages:
> [1] ggplot2_2.2.0  dplyr_0.5.0    sparklyr_0.5.3
> 
> loaded via a namespace (and not attached):
>  [1] Rcpp_0.12.10     withr_1.0.2      digest_0.6.12    rprojroot_1.2    assertthat_0.2.0 rappdirs_0.3.1  
>  [7] plyr_1.8.4       grid_3.3.2       R6_2.2.0         jsonlite_1.4     gtable_0.2.0     DBI_0.6-1       
> [13] backports_1.0.3  magrittr_1.5     scales_0.4.1     httr_1.2.1       lazyeval_0.2.0   config_0.2      
> [19] tools_3.3.2      munsell_0.4.3    parallel_3.3.2   yaml_2.1.14      colorspace_1.2-6 base64enc_0.1-3 
> [25] tibble_1.3.0

**Comment by yitao-li** [comment]
The support for stratified sampling (i.e., `dplyr::group_by(...)` followed by `dplyr::sample_n(...)` or `dplyr::sample_frac(...)`) has been shipped in sparklyr 1.5. I recommend installing the latest version of sparklyr to try it out.
